packages/keystone/src/fields/types/float/views/index.tsx

```diff
@@ -43,17 +163,40 @@ export const CardValue: CardValueComponent = ({ item, field }) => {
   );
 };
 
-export const controller = (config: FieldControllerConfig): FieldController<string, string> => {
+export const controller = (
+  config: FieldControllerConfig<{ validation: Validation; defaultValue: number | null }>
+): FieldController<Value, string> & {
+  validation: Validation;
+} => {
   return {
     path: config.path,
     label: config.label,
     graphqlSelection: config.path,
-    defaultValue: '',
-    deserialize: data => {
-      const value = data[config.path];
-      return typeof value === 'number' ? value + '' : '';
+    validation: config.fieldMeta.validation,
+    defaultValue: {
+      kind: 'create',
+      value: config.fieldMeta.defaultValue,
+    },
+    deserialize: data => ({
+      kind: 'update',
+      initial: data[config.path],
+      value: data[config.path],
+    }),
+    serialize: value => {
+      let vale = value.value;
+      let val;
+
+      if (vale === '' || vale === null) {
+        val = null;
+      } else if (typeof vale === 'string') {
+        val = parseFloat(vale);
+      } else {
+        val = vale;
+      }
+
+      return { [config.path]: val };
```

Comment:
It should already be parsed here
```suggestion
    serialize: value => {
      return { [config.path]: value.value };
```